test_rclcpp/CMakeLists.txt

```diff
@@ -69,6 +69,26 @@ if(AMENT_ENABLE_TESTING)
         "${middleware_impl}"
         "rclcpp")
     endif()
+
+    # test_timeout_subscriber
+    ament_add_gtest(
+      gtest_timeout_subscriber__${middleware_impl}
+      "test/test_timeout_subscriber.cpp"
+      TIMEOUT 30
+    )
+    if(TARGET gtest_timeout_subscriber__${middleware_impl})
+      target_link_libraries(gtest_timeout_subscriber__${middleware_impl}
+        ${_AMENT_EXPORT_ABSOLUTE_LIBRARIES}
+        ${_AMENT_EXPORT_LIBRARY_TARGETS}
+        ${GTEST_MAIN_LIBRARIES})
```

Comment:
You should not use `GTEST_MAIN_LIBRARIES` in combination with `ament_add_gtest`.